### PR TITLE
docs: Add details of project folder structure

### PR DESCRIPTION
**Pull request by merlinc**
## Proposed changes

Add some documentation about the folder structure of the app

### CODE REVIEW

File: docs/folder-structure.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      Nunjucks page templates
+    - `module/validators`
+    
+       Form validators
+       
+    - `module/middleware`
+    
+       Middleware helpers to decorate the request with specific module data
+
+- `artifacts/` **contains auto-generated files**
```
Comment (teneightfive) [inline docs/folder-structure.md:36]: Do you think we should document other directories we gitignore, like `.build/`?

**Review by teneightfive — APPROVED** [review]
This is AMAZING! Thank you 🙌

File: docs/folder-structure.md (hunk 1 above)
Comment (merlinc) [reply docs/folder-structure.md:47]: Added in the `.build` and .circleci` directories.

**Comment by solidgoldpig** [comment]
Great docs!

**Pull request merged**